src/simplify_algebra.cpp

```diff
@@ -406,7 +406,7 @@ struct find_splits
     auto matcher() const
     {
         return match::any(match::any_of[match::outputs()](
-            match::name("slice")(match::any_of[match::outputs()](pointwise()))));
+            match::name("slice")(match::any_of[match::outputs()](pointwise(), match::name_contains("reduce")))));
```

Comment:
Just like we have a `pointwise` matcher(defined [here](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/2ab73fd32145fa645da05c0e6e2fee77cd1889a0/src/simplify_algebra.cpp#L43)), could we add a `reduction` matcher to check for reduce operators?